openfe/setup/methods/openmm/equil_rbfe_methods.py

```diff
@@ -0,0 +1,711 @@
+            integrator_settings=IntegratorSettings(),
+            simulation_settings=SimulationSettings(
+                equilibration_length=2.0 * unit.nanosecond,
+                production_length=5.0 * unit.nanosecond,
+            )
+        )
+
+    def to_dict(self) -> dict:
+        """Serialize to dict representation"""
+        return {
+            'stateA': self._stateA.to_dict(),
+            'stateB': self._stateB.to_dict(),
+            'mapping': self._mapping.to_dict(),
+            'settings': dict(self._settings),
+        }
+
+    @classmethod
+    def from_dict(cls, d: dict):
+        """Deserialize from a dict representation"""
+        return cls(
+            stateA=ChemicalState.from_dict(d['stateA']),
+            stateB=ChemicalState.from_dict(d['stateB']),
+            mapping=LigandAtomMapping.from_dict(d['mapping']),
+            settings=dict(**d['settings']),
+        )
+
+    def run(self) -> bool:
+        stateA_openff_ligand = self._stateA.components['ligand'].to_openff()
+        stateB_openff_ligand = self._stateB.components['ligand'].to_openff()
+
+        smirnoff_stateA = SMIRNOFFTemplateGenerator(
+            forcefield=self._settings.topology_settings.forcefield['ligand'],
+            molecules=[stateA_openff_ligand],
+        )
+        omm_forcefield_stateA = app.ForceField(
+            *[ff for (comp, ff) in self._settings.topology_settings.forcefield.items()
+              if not comp == 'ligand']
+        )
+        omm_forcefield_stateA.registerTemplateGenerator(smirnoff_stateA.generator)
+
+        stateA_modeller = app.Modeller(
+            stateA_openff_ligand.to_topology().to_openmm(),
+            stateA_openff_ligand.conformers[0],
+        )
+
+        # Solvate the complex in a 0.15 mM cubic water box with 1.2 nm from the
+        # solute to the edges of the box
+        stateA_modeller.addSolvent(
+            omm_forcefield_stateA,
+            model=self._settings.topology_settings.solvent_model,
+            padding=self._settings.solvent_padding,
+            ionicStrength=self._stateA.components['solvent'].concentration,
+        )
+
+        # Create OpenMM system + topology + initial positions for "A" system
+        nonbonded_method = {
+            # TODO: Other methods here
+            'PME': app.PME,
+        }[self._settings.system_settings.nonbonded_method]
+        constraints = {
+            # TODO: Other constraints here
+            'HBonds': app.HBonds,
+        }[self._settings.system_settings.constraints]
+
+        stateA_system = omm_forcefield_stateA.createSystem(
+            stateA_modeller.topology,
+            nonbondedMethod=nonbonded_method,
+            nonbondedCutoff=self._settings.system_settings.nonbonded_cutoff,
+            constraints=constraints,
+            rigidWater=self._settings.system_settings.rigid_water,
+        )
+
+        stateA_topology = stateA_modeller.getTopology()
+
+        # TODO: Algorithm this magic number
```

Comment:
Should be just half the box length - I really need to work out _why_ this is so necessary...